### PR TITLE
feat(vault): re-introduce vault plugin

### PR DESCRIPTION
**Pull request by nervo**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.
- [x] If the code introduces new aliases, I provide a valid use case for all plugin users down below.

## Changes:

Re-introduce [Vault](https://www.vaultproject.io/) plugin.

## Other comments:

Previous vault plugin have been deprecated, then removed, because there is now an official completion. This new version exactly use this one :)

### CODE REVIEW

**Comment by Keramblock** [comment]
Just adding comment for the others encountered today the following error:
`[oh-my-zsh] plugin 'vault' not found`
Probably we need to wait for this PR

**Comment by nervo** [comment]
@Keramblock, sorry to ask, but are you sure you're on the good branch ?

**Comment by Keramblock** [comment]
@nervo I saw a message today and started looking for it in the zsh repo. I found this PR. I didn't change anything since I installed it, except for adding Powerlevel10k and some plugins on top of oh my zsh.

**Comment by nervo** [comment]
@Keramblock  I think i got it :)
When you say "Probably we need to wait for this PR", you don't mean "i have tested this pr and it does'nt work", you mean "i have not tested this pr, but it looks promising". Am i right ?

**Comment by Keramblock** [comment]
@nervo  yes, I think we need this PR for this message to disappear =)

File: plugins/vault/vault.plugin.zsh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Completion
+if (( ! $+commands[vault] )); then
+  return
+fi
+
+complete -o nospace -C vault vault
```
Comment (carlosala) [inline plugins/vault/vault.plugin.zsh:6]:
```suggestion
autoload -Uz bashcompinit && bashcompinit
complete -o nospace -C vault vault
```

**Review by carlosala — APPROVED** [review]
LGTM!

**Comment by carlosala** [comment]
Thanks for your contribution!

**Pull request merged**